src/Microsoft.AspNet.Mvc.Core/HttpResponseStreamWriter.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System.IO;
+using System.Text;
+using Microsoft.AspNet.Mvc.Internal;
+
+namespace Microsoft.AspNet.Mvc
+{
+    /// <summary>
+    /// Wraps the supplied <see cref="Stream"/> to prevent it from flushing (which can cause response to be sent
+    /// in Chunked encoding) and disposing (as the hosting layers own the stream).
+    /// Also wraps the supplied <see cref="Encoding"/> to prevent writing the preamble or BOM bytes to the response.
+    /// </summary>
+    public class HttpResponseStreamWriter : StreamWriter
```

Comment:
Right, I discussed with @Tratcher about this...but first we need to identify what are the issues that we are trying to solve...currently we have identified only the Flushing and BOM...like mainly if we are already seeing any issues with buffering and if that's something we need to solve...